network/setup-ip

```diff
@@ -1,135 +1,225 @@
+addresses1=$ip6;$prefix6;$gateway6"
+        fi
+    else
+        ip4_nm_config="${ip4_nm_config}
+addresses1=$ip;$prefix"
+        if [ -n "$ip6" ]; then
+            ip6_nm_config="${ip6_nm_config}
+addresses1=$ip6;$prefix6"
+        fi
+    fi
+    if [ -n "$ip4_nm_config" ]; then
+        cat >> "$nm_config" <<__EOF__
 [ipv4]
 method=manual
 may-fail=false
 $ip4_nm_config
 __EOF__
-            else
-                cat >> "$nm_config" <<__EOF__
+    else
+        cat >> "$nm_config" <<__EOF__
 [ipv4]
 method=ignore
 __EOF__
-            fi
+    fi
 
-            if [ -n "$ip6_nm_config" ]; then
-                cat >> "$nm_config" <<__EOF__
+    if [ -n "$ip6_nm_config" ]; then
+        cat >> "$nm_config" <<__EOF__
 [ipv6]
 method=manual
 may-fail=false
 $ip6_nm_config
 __EOF__
-            else
-                cat >> "$nm_config" <<__EOF__
+    else
+        cat >> "$nm_config" <<__EOF__
 [ipv6]
 method=ignore
 __EOF__
+    fi
+
+    chmod 600 "$nm_config"
+    # reload connection
+    nmcli connection load "$nm_config" || :
+}
+
+configure_qubes_ns() {
+    gateway=$(qubesdb-read /qubes-netvm-gateway)
+    #netmask=$(qubesdb-read /qubes-netvm-netmask)
+    primary_dns=$(qubesdb-read /qubes-netvm-primary-dns 2>/dev/null || echo "$gateway")
+    secondary_dns=$(qubesdb-read /qubes-netvm-secondary-dns)
+    echo "NS1=$primary_dns" > /var/run/qubes/qubes-ns
+    echo "NS2=$secondary_dns" >> /var/run/qubes/qubes-ns
+    /usr/lib/qubes/qubes-setup-dnat-to-ns
+}
+
+qubes_ip_change_hook() {
+    if [ -x /rw/config/qubes-ip-change-hook ]; then
+        /rw/config/qubes-ip-change-hook
+    fi
+    # XXX: Backward compatibility
+    if [ -x /rw/config/qubes_ip_change_hook ]; then
+        /rw/config/qubes_ip_change_hook
+    fi
+}
+
+have_qubesdb || exit 0
+
+if [ -n "$INTERFACE" ]; then
+    if [ "$ACTION" == "add" ]; then
+        MAC="$(get_mac_from_iface "$INTERFACE")"
+        if [ -n "$MAC" ]; then
+            ip="$(/usr/bin/qubesdb-read "/net-config/$MAC/ip" 2> /dev/null)"
+            ip6="$(/usr/bin/qubesdb-read "/net-config/$MAC/ip6" 2> /dev/null)"
+            netmask="$(/usr/bin/qubesdb-read "/net-config/$MAC/netmask" 2> /dev/null)"
+            netmask6="$(/usr/bin/qubesdb-read "/net-config/$MAC/netmask6" 2> /dev/null)"
+            gateway="$(/usr/bin/qubesdb-read "/net-config/$MAC/gateway" 2> /dev/null)"
+            gateway6="$(/usr/bin/qubesdb-read "/net-config/$MAC/gateway6" 2> /dev/null)"
+
+            # Handle legacy values
+            LEGACY_MAC="$(/usr/bin/qubesdb-read /qubes-mac 2> /dev/null)"
+            if [ "$MAC" == "$LEGACY_MAC" ] || [ "x$LEGACY_MAC" == "x" ]; then
+                if [ "x$ip" == "x" ]; then
+                    ip="$(/usr/bin/qubesdb-read /qubes-ip 2> /dev/null)"
+                fi
+                if [ "$ip6" == "x" ]; then
+                    ip6="$(/usr/bin/qubesdb-read /qubes-ip6 2> /dev/null)"
+                fi
+                if [ "$gateway" == "x" ]; then
```

Comment:
This lack "x" on the left side. Result: no default gateway set.
Shouldn't this be `[ -z "$gateway" ]` instead? (and the same for others)